lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -813,6 +815,18 @@ namespace {
         return finish(true, TC.resolveDeclRefExpr(unresolved, DC));
       }
 
+      // For method selectors with unresolved sub-expressions,
+      // resolve the expression but ignore local variables to allow
+      // conflicting method names
+      if (auto selector = dyn_cast<ObjCSelectorExpr>(expr)) {
+        if (selector->isMethodSelector()) {
```

Comment:
To limit the change to Swift 4, should I wrap lines 821 to 828 with, would this be sufficient?

``` c++
if !TC.getLangOptions().isSwiftVersion3() { 
  ...
}
```